src/ConstructiveSolidGeometry/SurfacePrimitives/EllipsoidMantle.jl

```diff
@@ -20,13 +20,46 @@ Surface primitive describing the surface of an [`Ellipsoid`](@ref).
 * `origin::CartesianPoint{T}`: Origin of the `Ellipsoid` which has this `EllipsoidMantle` as surface.
 * `rotation::SMatrix{3,3,T,9}`: Rotation matrix of the `Ellipsoid` which has this `EllipsoidMantle` as surface.
 """
-@with_kw struct EllipsoidMantle{T,TR,TP,TT,D} <: AbstractCurvedSurfacePrimitive{T}
-    r::TR = 1
-    φ::TP = nothing
-    θ::TT = nothing
+struct EllipsoidMantle{T,TR,TP,TT,D} <: AbstractCurvedSurfacePrimitive{T}
```

Comment:
Should we restrict `TP` and `TT` to `Nothing` here (as this is the only thing defined so far)?
And what about `TR` and `D`?

Reply:
`TP` and `TT` we could restrict for now. 

I would not restrict `TR` and `D`. In general, in Julia, you want to be as unrestricted as possible
to allow for possible compatibility between packages and their types.
(We should actually remove some restrictions at some point.)